gfx/video_crt_switch.c

```diff
@@ -180,6 +187,12 @@ static bool crt_sr2_init(videocrt_switch_t *p_switch, int monitor_index, unsigne
 
       RARCH_LOG("[CRT]: SR rtn %d \n", p_switch->rtn);
 
+      if(p_switch->rtn == 1)
+      {
+         p_switch->core_name = (char*)malloc(sizeof(char)*256); /* same size as library_name on retroarch_data.h */
```

Comment:
Is it guaranteed that `p_switch->core_name` will be `NULL` at this point?

For general safety, it would be nice to add a check here:

```c
   if (p_switch->core_name)
      free(p_switch->core_name)
   p_switch->core_name = (char*)malloc(sizeof(char)*256); /* same size as library_name on retroarch_data.h */
```

Reply:
No, there is no guarantee that it will be `null`. The safe assumption is that it will be pointing to a garbage address since it is never initialized. Upon further review, there is no reasonable location in the code to initialize it either, nor is it guaranteed that this only runs once; there is an assumption, based on the logic rules, that the `malloc` **has** to happen before the `free`, and that every `malloc` has a matching `free`, but I don't like to rely on this kind of assumptions either.

Since I don't want to do any big changes in the code (like adding some kind of initialization hook where I can ensure the pointer starts with a null value), and with the experience that came with the changes that came after this initial implementation, I've just changed `core_name` to be a static variable local to the .c file, just like `content_dir` is. So both variables are treated the same and follow the same rules. So no `malloc` or `free` happens anymore.

Changes are present on the just-amended commit.

Reply:
An excellent decision - this is a much safer approach.